unit test: nil pointer dereference in pkg/nodeupdater/utils_test.go

This unit test is failing with a nil pointer dereference:

```
$ make test
/home/jon/go/bin/gotestcover -v -race -short -coverprofile=cover.out github.com/IBM/vpc-node-label-updater/pkg/nodeupdater
command [go test -race -v -short -cover -coverprofile /tmp/gotestcover-69566083 github.com/IBM/vpc-node-label-updater/pkg/nodeupdater]: exit status 1
=== RUN   TestUpdateNodeLabel
    node_label_test.go:76: Test case: empty accessToken
    node_label_test.go:76: Test case: Empty riaasInstanceURL
    node_label_test.go:76: Test case: invalid riaasInstanceURL
--- PASS: TestUpdateNodeLabel (0.31s)
=== RUN   TestReadStorageSecretConfiguration
--- PASS: TestReadStorageSecretConfiguration (0.01s)
=== RUN   TestGetAccessToken
    utils_test.go:110: Test case: valid Request
--- FAIL: TestGetAccessToken (0.80s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x18b4afc]

goroutine 86 [running]:
testing.tRunner.func1.2({0x1983e80, 0x25efd00})
        /usr/local/go/src/testing/testing.go:1389 +0x366
testing.tRunner.func1()
        /usr/local/go/src/testing/testing.go:1392 +0x5d2
panic({0x1983e80, 0x25efd00})
        /usr/local/go/src/runtime/panic.go:844 +0x258
github.com/IBM/vpc-node-label-updater/pkg/nodeupdater.TestGetAccessToken(0xc0004beb60)
        /home/jon/workspace/ibm-vpc-node-label-updater/pkg/nodeupdater/utils_test.go:113 +0x61c
testing.tRunner(0xc0004beb60, 0x1b85808)
        /usr/local/go/src/testing/testing.go:1439 +0x214
created by testing.(*T).Run
        /usr/local/go/src/testing/testing.go:1486 +0x725
FAIL    github.com/IBM/vpc-node-label-updater/pkg/nodeupdater   1.159s
FAIL
test failed
make: *** [Makefile:58: test] Error 1
```

It's crashing on this line:
https://github.com/IBM/vpc-node-label-updater/blob/e5af1e5c88139a13d30c5632bb9bf638a4dad799/pkg/nodeupdater/utils_test.go#L113
This is because `https://iam.bluemix.net/oidc/token` returns an error, but [expErr is nil](https://github.com/IBM/vpc-node-label-updater/blob/e5af1e5c88139a13d30c5632bb9bf638a4dad799/pkg/nodeupdater/utils_test.go#L92) (not expected to return an error), which means tc.expErr.Error() results in a nil pointer dereference.

```
$ curl -s -I https://iam.bluemix.net/oidc/token | grep HTTP
HTTP/1.1 405 Method Not Allowed
```

It looks like that 405 response only results in an error from GetAccessToken due to this recent commit:
https://github.com/IBM/vpc-node-label-updater/commit/e5af1e5c88139a13d30c5632bb9bf638a4dad799#diff-9e3997c3319be0bdd79f52143d210dad803c412523bdf6e0bd39bd8bdbea3e5eR125-R130

So there are 2 problems I see:
1) We need to check tc.expErr != nil before dereferencing it, so we can see what was returned intead of crashing the unit test.
2) We need to fix the "valid request" test case. Obviously we can't add valid credentials to the unit test, so perhaps that endpoint needs to be mocked out somehow.

/cc @prankulmahajan @ambiknai @arahamad

--- a/pkg/nodeupdater/utils_test.go
+++ b/pkg/nodeupdater/utils_test.go
@@ -109,7 +109,7 @@ func TestGetAccessToken(t *testing.T) {
 	for _, tc := range testCases {
 		t.Logf("Test case: %s", tc.name)
 		_, err := tc.secretConfig.GetAccessToken(logger)
-		if err != nil {
+		if err != nil && tc.expErr != nil {
 			if err.Error() != tc.expErr.Error() && !strings.Contains(err.Error(), tc.expErr.Error()) {
 				t.Fatalf("Expected error code: %v, got: %v. err : %v", tc.expErr, err, err)
 			}
@@ -199,6 +199,7 @@ func TestCheckIfRequiredLabelsPresent(t *testing.T) {
 	exp := CheckIfRequiredLabelsPresent(labelMap)
 	assert.Equal(t, exp, false)
 	labelMap[vpcBlockLabelKey] = "true"
+	labelMap[instanceIDLabelKey] = "true"
 	ex := CheckIfRequiredLabelsPresent(labelMap)
 	assert.Equal(t, ex, true)
 }

